src/Filesystem/Dir.php

```diff
@@ -220,143 +220,145 @@ public static function inventory(
-						$content[] = pathinfo($item, PATHINFO_FILENAME);
-						break;
-					default:
-						$inventory['files'][$item] = [
-							'filename'  => $item,
-							'extension' => $extension,
-							'root'      => $root,
-						];
-				}
+			$extension = pathinfo($item, PATHINFO_EXTENSION);
+
+			// don't track files with these extensions
+			if (in_array($extension, ['htm', 'html', 'php']) === true) {
+				continue;
 			}
-		}
 
-		// remove the language codes from all content filenames
-		if ($multilang === true) {
-			foreach ($content as $key => $filename) {
-				$content[$key] = pathinfo($filename, PATHINFO_FILENAME);
+			// collect all content files separately,
+			// not as inventory entries
+			if ($extension === $contentExtension) {
+				$filename = pathinfo($item, PATHINFO_FILENAME);
+
+				// remove the language codes from all content filenames
+				if ($multilang === true) {
+					$filename = pathinfo($filename, PATHINFO_FILENAME);
+				}
+
+				$content[] = $filename;
+				continue;
 			}
 
-			$content = array_unique($content);
+			// collect all other files
+			$inventory['files'][$item] = [
+				'filename'  => $item,
+				'extension' => $extension,
+				'root'      => $root,
+			];
 		}
 
-		$inventory = static::inventoryContent($inventory, $content);
-		$inventory = static::inventoryModels($inventory, $contentExtension, $multilang);
+		$content = array_unique($content);
+
+		$inventory['template'] = static::inventoryTemplate(
+			$content,
+			$inventory['files']
+		);
 
 		return $inventory;
 	}
 
 	/**
-	 * Take all content files,
-	 * remove those who are meta files and
-	 * detect the main content file
+	 * Collect information for a child for the inventory
 	 */
-	protected static function inventoryContent(array $inventory, array $content): array
-	{
-		// filter meta files from the content file
-		if (empty($content) === true) {
-			$inventory['template'] = 'default';
-			return $inventory;
+	protected static function inventoryChild(
+		string $item,
+		string $root,
+		string $contentExtension = 'txt',
+		bool $multilang = false
+	): array {
+		// extract the slug and num of the directory
+		if ($separator = strpos($item, static::$numSeparator)) {
+			$num  = (int)substr($item, 0, $separator);
+			$slug = substr($item, $separator + 1);
 		}
 
-		foreach ($content as $contentName) {
-			// could be a meta file. i.e. cover.jpg
-			if (isset($inventory['files'][$contentName]) === true) {
-				continue;
+		// determine the model
+		if (empty(Page::$models) === false) {
+			if ($multilang === true) {
+				$code = App::instance()->defaultLanguage()->code();
+				$contentExtension = $code . '.' . $contentExtension;
 			}
 
-			// it's most likely the template
-			$inventory['template'] = $contentName;
+			// look if a content file can be found
+			// for any of the available models
+			foreach (Page::$models as $modelName => $modelClass) {
+				if (file_exists($root . '/' . $modelName . '.' . $contentExtension) === true) {
```

Comment:
I think we should use `is_file()` instead, otherwise it will also be true for folders with that name.